channel.go

```diff
@@ -909,19 +909,15 @@ func toStringSet(ss []string) map[string]struct{} {
 }
 
 // take a list of service::method formatted string and make
-// the map[service]map[service::method]struct{} set
-func toServiceMethodSet(sms []string) (map[string]map[string]struct{}, error) {
-	set := map[string]map[string]struct{}{}
+// the map[service::method]struct{} set
+func toServiceMethodSet(sms []string) (map[string]struct{}, error) {
+	set := map[string]struct{}{}
 	for _, sm := range sms {
 		s := strings.Split(sm, "::")
 		if len(s) != 2 {
 			return nil, fmt.Errorf("each %q value should be of service::Method format but got %q", "SkipHandlerMethods", sm)
 		}
-		svc := s[0]
-		if _, ok := set[svc]; !ok {
-			set[svc] = map[string]struct{}{}
-		}
-		set[svc][sm] = struct{}{}
+		set[sm] = struct{}{}
```

Comment:
nit we can `if s:= .. ; len(s) !=2` here now :)

Reply:
(optional suggestion) Do we actually care about `sm` validation here, since `svc := s[0]` is out?

Neither `service` or `method` name normalization, nor validation of service/method existence seem to be happening for `sm`.  If they don't have to be correct, why even check for `::` format?

Reply:
@bryndin, I would prefer to keep the check to catch incorrectly configured method names, where the method name must always contain the thrift service prefix.

Reply:
yea I agree with @jronak. This is the bare minimal validation we can do to make sure it is X::X format.

Reply:
Agree.
Customers asked us to add full validation on service/method spelling, they had outages due to accidental typos. If we ever do that, we'd refactor this code.